src/grammar/grammar.ohm

```diff
@@ -173,10 +173,16 @@ Tact {
     // Integer Literal
     // hexDigit defined in Ohm's built-in rules (otherwise: hexDigit = "0".."9" | "a".."f" | "A".."F")
     // digit defined in Ohm's built-in rules (otherwise: digit = "0".."9")
-    integerLiteral = integerLiteralHex | integerLiteralDec // Order is important
-    integerLiteralDec = digit+
-    integerLiteralHex = "0x" hexDigit+
-                      | "0X" hexDigit+
+    integerLiteral = integerLiteralHex | integerLiteralBin | integerLiteralOct | integerLiteralDec // Order is important
+    integerLiteralDec = digit+ ("_" | digit)*
+    integerLiteralHex = "0x" hexDigit+ ("_" | hexDigit)*
+                      | "0X" hexDigit+ ("_" | hexDigit)*
+    integerLiteralBin = "0b" binDigit+ ("_" | binDigit)*
+                      | "0B" binDigit+ ("_" | binDigit)*
+    integerLiteralOct = "0o" octDigit+ ("_" | octDigit)*
+                      | "0O" octDigit+ ("_" | octDigit)*
```

Comment:
I'd model literals after JS/TS, because it's kind of a go-to language for TON, so people could copy some literals from Tact code into TS tests or something like that.

There is one caveat though: JS's octal literals starting with `0`. For the reasons of backwards compatibility it might be a bit too late to not allow integer literals with leading zeros. So this might result in some suboptimal UX when copying things from Tact to TS tests. (We could introduce a warning about this, though).

These are JS's restrictions, regarding underscores in numeric literals:
1. JS does not allow `_` at the end of literals.
2. As mentioned earlier, we should also forbid `_` after leading zeros.
3. JS does not allow multiple `_` in a row (so `4_2` is a valid literal, but `4__2` is not).
 
Of course, some other languages, like [OCaml](https://ocaml.org) are very liberal in this respect, e.g. it's ok with `0004____2_` being a valid integer literal. However, I think going with JS's grammar (except for octals) is something we should do.